src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -33,11 +35,26 @@ protected ExternalWorkspace run() throws Exception {
         if (step.getUpstream() == null) {
             // this is the upstream job
 
-            DiskAllocationStrategy allocationStrategy = new MostUsableSpaceStrategy(step.getDiskPoolId(), step.getDescriptor().getDiskPools());
+            String diskPoolId = step.getDiskPoolId();
+            if (diskPoolId == null) {
+                throw new Exception("Disk Pool ID was not provided as step parameter");
+            }
+
+            List<DiskPool> diskPools = step.getDescriptor().getDiskPools();
+            DiskAllocationStrategy allocationStrategy = new MostUsableSpaceStrategy(diskPoolId, diskPools);
             Disk disk = allocationStrategy.allocateDisk();
 
-            String pathOnDisk = computePathOnDisk(disk.getPhysicalPathOnDisk());
-            ExternalWorkspace externalWorkspace = new ExternalWorkspace(disk.getDiskId(), pathOnDisk);
+            String physicalPathOnDisk = disk.getPhysicalPathOnDisk();
+            if (physicalPathOnDisk == null) {
+                throw new Exception("Physical path on disk was not provided in the Jenkins global config");
```

Comment:
Yes indeed, this can be optional.
So maybe, if this parameter is not provided, I could fallback to a default path, e.g.: "workspace". And maybe display an info message. WDYT?